pkg/kubecost/window.go

```diff
@@ -410,6 +410,32 @@ func (w Window) Expand(that Window) Window {
 	return w
 }
 
+func (w Window) ContractStart(start time.Time) Window {
+	if w.start == nil || start.After(*w.start) {
+		w.start = &start
+	}
+	return w
+}
+
+func (w Window) ContractEnd(end time.Time) Window {
+	if w.end == nil || end.Before(*w.end) {
+		w.end = &end
+	}
+	return w
+}
+
+func (w Window) Contract(that Window) Window {
+	if that.start != nil {
+		w = w.ContractStart(*that.start)
+	}
+
+	if that.end != nil {
+		w = w.ContractEnd(*that.end)
+	}
```

Comment:
Why are we avoiding contracting if start/end is `nil`? `nil` implies an open (infinite?) window, right? I think the `ContractStart` and `ContractEnd` already handle this well.

Reply:
The idea is that if `that.start` or `that.end` is `nil` then there's no value of `w` that  could possibly be contracted (`nil` is, in that sense, the "max") so we don't have any actions to take. Is that fair to say? I want to write some unit tests for this when I have a chance!

Reply:
Oh! I read this wrong, I thought we were checking if `w.start != nil`, whoops! Makes total sense!